CMakeLists.txt

```diff
@@ -26,11 +26,28 @@ set(OPENEXR_VERSION_RELEASE_TYPE "-dev" CACHE STRING "Extra version tag string f
 set(OPENEXR_VERSION ${OpenEXR_VERSION})
 set(OPENEXR_VERSION_API "${OpenEXR_VERSION_MAJOR}_${OpenEXR_VERSION_MINOR}")
 
+# Library/shared-object version using libtool versioning policy.
 # See https://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html
-set(OPENEXR_SOVERSION 29)
-set(OPENEXR_SOAGE 0) 
-set(OPENEXR_SOREVISION 0) 
-set(OPENEXR_LIB_VERSION "${OPENEXR_SOVERSION}.${OPENEXR_SOREVISION}.${OPENEXR_SOAGE}")
+#
+# Library API version (CMake's library VERSION attribute) is of the
+# form CURRENT.REVISION.AGE; the CMake SOVERSION attribute corresonds
+# to just CURRENT. These produce a .so and a symlink symlink, e.g.:
+# libOpenEXR-3_1.so.29 -> libOpenEXR-3_1.so.29.0.0
+#                    ^                       ^ ^ ^
+#                    |                       | | |
+#                    CURRENT                 | | AGE
+#                                            | REVISION 
+#                                            CURRENT
+# When updating:
+#   1. no API change: CURRENT.REVISION+1.AGE
+#   2. API added:     CURRENT+1.0.AGE+1
+#   3. API changed:   CURRENT+1.0.0
+#
+set(OPENEXR_LIBTOOL_CURRENT 29)
+set(OPENEXR_LIBTOOL_REVISION 0)
+set(OPENEXR_LIBTOOL_AGE 0)
+set(OPENEXR_LIB_VERSION "${OPENEXR_LIBTOOL_CURRENT}.${OPENEXR_LIBTOOL_REVISION}.${OPENEXR_LIBTOOL_AGE}")
```

Comment:
Hear me out on this kooky idea:

Our version numbers are MAJOR.MINOR.PATCH. We assume that even minor releases change API, but that we don't change API for patch releases, right?

What if CURRENT = MAJOR*100+MINOR, and REVISION=0 (by default), and AGE = PATCH? Then the vast majority of time, we wouldn't need to adjust anything at all for a release. And very very occasionally, if a patch release necessitated adding an API call, we would bump AGE in that release branch.

The advantage to making this automated based on the release version is that it's much harder to make mistakes. If these numbers are all varying independently of release version and changed by hand, we have to be much more careful about bumping soversion on every release and doing it *correctly*, remembering which kinds of changes got incorporated into that release. Frankly, I don't trust us to do it right, unless we have the CI include a fully automated and foolproof test that confirms that API doesn't change without these numbers being bumped properly.

Reply:
I said it wrong. I meant, 

What if CURRENT = MAJOR*100+MINOR, and REVISION=0 (by default), and AGE = PATCH? Then the vast majority of time, we wouldn't need to adjust anything at all for a release. And very very occasionally, if a patch release necessitated adding an API call, we would bump **REVISION** in that release branch.

Reply:
Furthermore, since the name of the library contains the major and minor releases, CURRENT can always be 0 for all we care.

Reply:
Wait, that's not quite true. If we ever have a patch release that adds an API call (do we? would we?), then we would need to bump current. I find this scheme maddeningly easy to misunderstand or to do it incorrectly.

Reply:
The 2.5.x series is in the category, isn't it? Patch releases with API changes.